### PR TITLE
default_link API and supporting changes

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

This PR adds an API for computing the "default_link" of an ArticulatedObject as well as caching and retrieving it from metadata.

Context: we want to be able to "open" an ArticulatedObject without specifying a link or opening all the links. 
The "default" link should be one which is accessible, so we choose the bottom-most link with a "revolute" or "prismatic" joint type. 
To accommodate customization we also allow this value to be set in the ao_config.json and read from metadata instead.

Example of using default_link to open/close an AO:
```
default_link = sutils.get_ao_default_link(sim, my_articulated_object, compute_if_not_found=True)
sutils.open_link(my_articulated_object, default_link)
```

NOTE: additional supporting changes include:
- `nan` checking for navmesh point snap in `unoccluded_navmesh_snap` method.
- refactored `get_excluded_recs_from_filter_file` to allow specific filter subset as input
- added `max_collision_depth` variable to `snap_down` method. Replaces hardcoded value and changes the default from `0.05` to `0.01` resulting in less overall penetration between clutter objects and support surfaces.

This PR also adds a documentation page to the habitat-lab website describing the taxonomy of "user_defined" metadata fields in the lab stack.

## How Has This Been Tested

Added a unit test. It validates the functionality as expected.

**NOTE: found a habitat-sim bug failing the unit test. We need to fix that, but it can wait until after merge. Unit test has TODO for this.**

## Types of changes

- **\[Docs change\]** Addition or changes to the documentation
- **\[Refactoring\]** Large changes to the code that improve its functionality or performance
- **\[Bug Fix\]** (non-breaking change which fixes an issue)
- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes if required.

### CODE REVIEW

**Comment by aclegg3** [comment]
Regarding the habitat-sim bug: it is not necessary to fix the bug for this feature to be merged. The bug manifests because I want to test all possible ways of setting the metadata. The expected paths (instance setting and config loading) work fine.

File: docs/pages/metadata-taxonomy.rst
Comment (0mdc) [inline docs/pages/metadata-taxonomy.rst:1]: I love this doc!

File: test/test_sim_utils.py [hunk 1]
```diff
@@ -522,6 +522,56 @@ def test_ao_open_close_queries():
...
+        )
+        assert default_link == 1
+        new_default_link = sutils.get_ao_default_link(
+            new_fridge, compute_if_not_found=True
+        )
+        print(
+            f" new_default_link (== {new_default_link}) should be 0, waiting on sim bug fix."
+        )
+        # TODO: habitat-sim bug. "default_link" does not get copied over after instantiation if set in the template programmatically.
+        # assert new_default_link == 0
```
Comment (0mdc) [inline test/test_sim_utils.py:565]: 👍

**Review by 0mdc — APPROVED** [review]
LGTM!

File: test/test_sim_utils.py (hunk 1 above)
Comment (jturner65) [reply test/test_sim_utils.py:565]: This issue is addressed by [this Habitat-Sim PR](https://github.com/facebookresearch/habitat-sim/pull/2359)

**Pull request merged**